Foundation/Boxing.swift

```diff
@@ -75,7 +75,9 @@ internal protocol _SwiftNativeFoundationType : class {
     
     func mutableCopy(with zone : NSZone) -> Any
     
+    func hash(into hasher: inout Hasher)
```

Comment:
In protocol extensions that forward hashing to some core value, it is generally a good idea to implement both `hash(into:)` and `hashValue`, so that hashing always matches the behavior of the underlying value. (But if we have to choose one, I'd go with just `hash(into:)`.